### ISSUE TITLE
Consider using @inlineable / @usableFromInline

### ISSUE THREAD

**Issue by thomasvl** [apple/swift-protobuf#763]
The Swift forums have had some threads about performance, especially on cross module things (like [_Swift’s unpredictable efficiency_](https://forums.swift.org/t/swifts-unpredictable-efficiency/13506)), and there have been posts from other library authors having had to use some of these to maintain performance (SwiftNIO might have?).

In a few places, the generated code is using small functions to hopefully get code inlined to and the real work done within the runtime support code.  So… should some of our generated code being using these annotations so call sites still get "fast" code even if the generated proto code goes into a different module from the call site?

**Comment by tbkka** [comment]
@inlineable can also expand the API surface significantly if used carelessly.  But it likely could be useful in a number of places.

**Comment by thomasvl** [comment]
@tbkka yup.  I just wonder if a lot of your past performance work was based on things being in a single module, so we might have a very different profile using a more recent swift and with some modules in the mix.  ☹️
